File: desktop/src/main/marketplace-config-store.ts
/**
 * Phase 3c: Per-entry config storage for marketplace packages.
 * Reads/writes ~/.claude/youcoded-config/<id>.json.
 *
 * This is separate from the plugin's own config (if any) and from the
 * unified youcoded-skills.json. Each marketplace entry that declares
 * a configSchema gets its own JSON file keyed by entry id.
 */

import fs from 'fs';
import path from 'path';
import os from 'os';

const CONFIG_DIR = path.join(os.homedir(), '.claude', 'youcoded-config');

/**
 * Get the config values for a marketplace entry.
 * Returns an empty object if no config has been saved yet.
 */
export function getConfig(id: string): Record<string, unknown> {
  try {
    const filePath = path.join(CONFIG_DIR, `${id}.json`);
    if (!fs.existsSync(filePath)) return {};
    const raw = fs.readFileSync(filePath, 'utf8');
    return JSON.parse(raw);
  } catch {
    return {};
  }
}

/**
 * Save config values for a marketplace entry.
 * Creates the config directory if it doesn't exist.
 */
export function setConfig(id: string, values: Record<string, unknown>): void {
  if (!fs.existsSync(CONFIG_DIR)) {
    fs.mkdirSync(CONFIG_DIR, { recursive: true });
  }
  // Atomic write: write to temp then rename
  const filePath = path.join(CONFIG_DIR, `${id}.json`);
  const tmpPath = filePath + '.tmp';
  fs.writeFileSync(tmpPath, JSON.stringify(values, null, 2), 'utf8');
  fs.renameSync(tmpPath, filePath);
}

/**
 * Delete config for a marketplace entry (used on uninstall).
 */
export function deleteConfig(id: string): void {
  try {
    const filePath = path.join(CONFIG_DIR, `${id}.json`);
    if (fs.existsSync(filePath)) fs.unlinkSync(filePath);
  } catch {
    // best-effort
  }
}

File: app/src/main/kotlin/com/youcoded/app/skills/MarketplaceConfigStore.kt
package com.youcoded.app.skills

import org.json.JSONObject
import java.io.File

/**
 * Phase 3c: Per-entry config storage for marketplace packages.
 * Reads/writes ~/.claude/youcoded-config/<id>.json.
 *
 * Separate from the plugin's own config and from youcoded-skills.json.
 * Each marketplace entry that declares a configSchema gets its own file.
 */
class MarketplaceConfigStore(private val homeDir: File) {

    private val configDir: File get() = File(homeDir, ".claude/youcoded-config")

    /** Get the config values for a marketplace entry. */
    fun getConfig(id: String): JSONObject {
        return try {
            val file = File(configDir, "$id.json")
            if (file.exists()) JSONObject(file.readText()) else JSONObject()
        } catch (_: Exception) {
            JSONObject()
        }
    }

    /** Save config values for a marketplace entry. */
    fun setConfig(id: String, values: JSONObject) {
        configDir.mkdirs()
        val file = File(configDir, "$id.json")
        val tmp = File(file.absolutePath + ".tmp")
        tmp.writeText(values.toString(2))
        tmp.renameTo(file)
    }

    /** Delete config for a marketplace entry (used on uninstall). */
    fun deleteConfig(id: String) {
        try {
            val file = File(configDir, "$id.json")
            if (file.exists()) file.delete()
        } catch (_: Exception) {
            // best-effort
        }
    }
}
